plugins/aws/ecs_helper.go

```diff
@@ -465,11 +474,10 @@ func bindClusters(clusterTag string, state ecsState, tmpls []containerBindTempla
 					continue
 				}
 
-				// ...with an IP address
-				ec2Host := ptr.StringValue(ec2inst.PrivateIpAddress)
-
 				// grab the container our template is for out of the task instance
 				container := tinst.getContainer(tmpl.container)
+				// ...with an IP address
+				HostIP := getIP(ec2inst,container)
```

Comment:
@vaibhavzo better comments over here to clarify why and when this would be needed?